docker/lib/dependabot/docker/utils/credentials_finder.rb

```diff
@@ -47,29 +47,33 @@ def build_aws_credentials(registry_details)
           # If credentials have been generated from AWS we can just return them
           return registry_details if registry_details["username"] == "AWS"
 
-          # If we don't have credentials, we might get them from the proxy
-          return registry_details if registry_details["username"].nil?
-
-          # Otherwise, we need to use the provided Access Key ID and secret to
-          # generate a temporary username and password
+          # Build a client either with explicit creds or default creds
+          registry_hostname = registry_details.fetch("registry")
+          region = registry_hostname.match(AWS_ECR_URL).named_captures.fetch("region")
           aws_credentials = Aws::Credentials.new(
             registry_details["username"],
             registry_details["password"]
           )
 
-          registry_hostname = registry_details.fetch("registry")
-          region = registry_hostname.match(AWS_ECR_URL).
-                   named_captures.fetch("region")
+          ecr_client =
+            if aws_credentials.set?
+              Aws::ECR::Client.new(region: region, credentials: aws_credentials)
+            else
+              # Let the client check default locations for credentials
+              Aws::ECR::Client.new(region: region)
```

Comment:
This is where we take a small perf hit in the Dependabot-on-GitHub service since it looks for config files in a number of places that will never exist... but I think those calls should be quite quick such that it doesn't really matter?

Reply:
I think this is fine, I guess the only mild risk is that it encounters a file-system related issue trying to access a certain path. It does make me think we should have some kind of environment flag that explicitly tells code we are running in the service and to skip certain behaviours like this.

I don't think of that as a performance requirement so much as reducing logic a job could 'snag' on due to unforeseen circumstances ( e.g. for some weird reason we end up needing to have .aws dotfiles in our image, or they get added by accident, etc, etc ).

Reply:
> I don't think of that as a performance requirement so much as reducing logic a job could 'snag' on due to unforeseen circumstances ( e.g. for some weird reason we end up needing to have .aws dotfiles in our image, or they get added by accident, etc, etc ).

That's a good point.

I'd prefer to ship this as-is for now as I don't expect any breakage and we can test-deploy before merge. And then have a follow-up ticket to add that flag... Also because I'm interested in doing a DRY audit across updater, dry-run, dependabot-script etc to tease out the common bits so adding that flag / env var would be easier to handle as part of that audit/re-think.